### PR TITLE
ENG-12892: Voltadmin Dr Reset with --cluster not guarding against loc…

### PR DESCRIPTION
**Pull request by JadenW**
…al clusterID and not connected cluster

### CODE REVIEW

File: lib/python/voltcli/voltadmin.d/dr.py [hunk 1]
```diff
@@ -80,10 +85,47 @@ def dr(runner):
...
     if runner.opts.clusterId >= 0 and runner.opts.resetAll:
         runner.abort_with_help('You cannot specify both --cluster and --all options.')
-    if runner.opts.clusterId < -1 or runner.opts.clusterId > 127:
-        runner.abort_with_help('The cluster ID must be in the range of 0 to 127.')
     if runner.opts.modifier == "drop" and (runner.opts.forcing or runner.opts.clusterId >=0 or runner.opts.resetAll):
         runner.abort_with_help('You cannot specify either --force, --cluster or --all for drop.')
     if runner.opts.timeout < 0:
         runner.abort_with_help('The timeout value must be non-negative seconds.')
+    if runner.opts.clusterId < -1 or runner.opts.clusterId > 127:
+        runner.abort_with_help('The remote-cluster-ID must be in the range of 0 to 127.')
```
Comment (slowsoul) [inline lib/python/voltcli/voltadmin.d/dr.py]: I think the "-"s should be removed

File: lib/python/voltcli/voltadmin.d/dr.py [hunk 2]
```diff
@@ -80,10 +85,47 @@ def dr(runner):
...
+    return int(host.clusterid)
+
+def getRemoteClusterIds(runner):
+    response = checkstats.get_stats(runner, "DRROLE")
+    clusterIds = []
+    for tuple in response.table(0).tuples():
+        remote_cluster_id = tuple[2]
+        if remote_cluster_id != -1:
+            clusterIds.append(remote_cluster_id)
+    return clusterIds
```
Comment (slowsoul) [inline lib/python/voltcli/voltadmin.d/dr.py]: Better to have a newline at the end

File: lib/python/voltcli/voltadmin.d/dr.py [hunk 3]
```diff
@@ -37,6 +41,7 @@ def drop(runner):
         runner.info(message)
     else:
         runner.error(message)
+        sys.exit(1)
```
Comment (slowsoul) [inline lib/python/voltcli/voltadmin.d/dr.py]: This may cause conflict with ENG-12939 branch

File: lib/python/voltcli/voltadmin.d/dr.py (hunk 1 above)
Comment (manjujames) [inline lib/python/voltcli/voltadmin.d/dr.py]: Right. Also from help string.

File: lib/python/voltcli/voltadmin.d/dr.py [hunk 4]
```diff
@@ -80,10 +85,47 @@ def dr(runner):
...
-    if runner.opts.clusterId < -1 or runner.opts.clusterId > 127:
-        runner.abort_with_help('The cluster ID must be in the range of 0 to 127.')
     if runner.opts.modifier == "drop" and (runner.opts.forcing or runner.opts.clusterId >=0 or runner.opts.resetAll):
         runner.abort_with_help('You cannot specify either --force, --cluster or --all for drop.')
     if runner.opts.timeout < 0:
         runner.abort_with_help('The timeout value must be non-negative seconds.')
+    if runner.opts.clusterId < -1 or runner.opts.clusterId > 127:
+        runner.abort_with_help('The remote-cluster-ID must be in the range of 0 to 127.')
+    if runner.opts.clusterId > 0 and runner.opts.clusterId == getOwnClusterId(runner):
+        runner.abort_with_help('The remote-cluster-ID must be different from local cluster ID.')
```
Comment (manjujames) [inline lib/python/voltcli/voltadmin.d/dr.py]: May be ask Andrew to look through all of these user facing messages?

**Pull request merged**